tasks/acs-config.yaml

```diff
@@ -0,0 +1,164 @@
+# Here is the priority of roles that should be applied to users:
+#   Admin > MISO security role > PO/TL role
+
+# Admin users with OpenShift auth provider (openshift):
+# 1 - Create if not exist:
+- name: Create rolebinding if not exists for OpenShift auth provider
+  uri:
+    url: "https://{{ api_endpoint }}/v1/groups"
+    method: POST
+    validate_certs: no
+    headers:
+        authorization: "{{ authorization }}"
+    body_format: json
+    body:
+      roleName: "{{ item.role_name }}"
+      props:
+        authProviderId: "{{ auth_provider_openshift_id }}"
+        key: "{{ item.key }}"
+        value: "{{ item.value }}"
+    status_code: [200,500]
+  with_items: "{{ custom_groups_with_openshift }}"
+
+# 2 - Update to make sure the new roles are applied:
+- name: Map role in OpenShift auth provider
+  uri:
+    url: "https://{{ api_endpoint }}/v1/groupsbatch"
+    method: POST
+    validate_certs: no
+    headers:
+        authorization: "{{ authorization }}"
+    body_format: json
+    body:
+        previous_groups:
+        - props:
+            authProviderId: "{{ auth_provider_openshift_id }}"
+            key: "{{ item.key }}"
+            value: "{{ item.value }}"
+        required_groups:
+        - roleName: "{{ item.role_name }}"
+          props:
+            authProviderId: "{{ auth_provider_openshift_id }}"
+            key: "{{ item.key }}"
+            value: "{{ item.value }}"
+    status_code: [200]
+  with_items: "{{ custom_groups_with_openshift }}"
+
+
+# MISO users with Azure AD auth provider (oidc):
+# 1 - Create if not exist:
+- name: Create rolebinding if not exists for Azure AD auth provider
+  uri:
+    url: "https://{{ api_endpoint }}/v1/groups"
+    method: POST
+    validate_certs: no
+    headers:
+        authorization: "{{ authorization }}"
+    body_format: json
+    body:
+      roleName: "{{ item.role_name }}"
+      props:
+        authProviderId: "{{ auth_provider_oidc_id }}"
+        key: "{{ item.key }}"
+        value: "{{ item.value }}"
+    status_code: [200,500]
+  with_items: "{{ custom_groups_with_azuread }}"
+
+# 2 - Update to make sure the new roles are applied:
+- name: Map role in Azure AD auth provider
+  uri:
+    url: "https://{{ api_endpoint }}/v1/groupsbatch"
+    method: POST
+    validate_certs: no
+    headers:
+        authorization: "{{ authorization }}"
+    body_format: json
+    body:
+      previous_groups:
+      - props:
+          authProviderId: "{{ auth_provider_oidc_id }}"
+          key: "{{ item.key }}"
+          value: "{{ item.value }}"
+      required_groups:
+      - roleName: "{{ item.role_name }}"
+        props:
+          authProviderId: "{{ auth_provider_oidc_id }}"
+          key: "{{ item.key }}"
+          value: "{{ item.value }}"
+    status_code: [200]
+  with_items: "{{ custom_groups_with_azuread }}"
+
+# Revoke the API token when done:
+- include_tasks: revoke_api_token.yaml
```

Comment:
Awesome that you revoke your token when done. One minor suggestion would be to make sure it runs even if there is an error elsewhere. https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html#handling-errors-with-blocks has docs on an `always` block that ensures tasks will run even if there is a failure elsewhere.